src/patito/polars.py

```diff
@@ -136,7 +136,7 @@ def derive(self: LDF, columns: list[str] | None = None) -> LDF:
         """
         derived_columns = []
         props = self.model._schema_properties()
-        original_columns = set(self.columns)
+        original_columns = set(self.collect_schema().names())
```

Comment:
This will be more efficient:

```suggestion
        original_columns = set(self.collect_schema())
```